cpp/tests/test_abm_world.cpp

```diff
@@ -282,6 +283,35 @@ TEST(TestWorld, evolveMigration)
         EXPECT_EQ(home.get_number_persons(), 2);
         EXPECT_EQ(hospital.get_number_persons(), 1);
     }
+
+    // Test if a dead person can still migrate
+    {
+        auto t      = mio::abm::TimePoint(0) + mio::abm::hours(8);
+        auto dt     = mio::abm::hours(11);
+        auto params = mio::abm::GlobalInfectionParameters{};
+        params.get<mio::abm::CarrierToInfected>()[{mio::abm::VirusVariant::Wildtype, mio::abm::AgeGroup::Age60to79,
+                                                   mio::abm::VaccinationState::Unvaccinated}]  = 0.1;
+        params.get<mio::abm::CarrierToRecovered>()[{mio::abm::VirusVariant::Wildtype, mio::abm::AgeGroup::Age60to79,
+                                                    mio::abm::VaccinationState::Unvaccinated}] = 0.1;
+
+        auto world       = mio::abm::World(params);
+        auto home_id     = world.add_location(mio::abm::LocationType::Home);
+        auto work_id     = world.add_location(mio::abm::LocationType::Work);
+        auto icu_id      = world.add_location(mio::abm::LocationType::ICU);
+
+        auto& p_dead = add_test_person(world, icu_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t);
+        p_dead.set_assigned_location(home_id);
+        p_dead.set_assigned_location(work_id);
+
+        mio::abm::TripList& data = world.get_trip_list();
+        mio::abm::Trip trip1(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(9), work_id, home_id);
+        mio::abm::Trip trip2(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(10), icu_id, home_id);
```

Comment:
```suggestion
        mio::abm::Trip trip2(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(10), home_id, icu_id);
```

The person is currently in ICU so a better test would be to test if he/she goes from ICU to home, not from home to ICU, right?

Reply:
You're correct. I didn't notice the order of destination and origin in Trip. It's fixed!

Reply:
Yes, that would be a better choice. Although these trips should not happen anyway since the person is dead.